app/src/main/java/org/oppia/android/app/home/promotedlist/PromotedStoryListViewModel.kt

```diff
@@ -8,37 +8,64 @@ import androidx.lifecycle.ViewModel
 import org.oppia.android.R
 import org.oppia.android.app.home.HomeItemViewModel
 import org.oppia.android.app.home.RouteToRecentlyPlayedListener
+import org.oppia.android.app.model.RecommendedActivityList
 import java.util.Objects
 
 /** [ViewModel] for the promoted story list displayed in [HomeFragment]. */
 class PromotedStoryListViewModel(
   private val activity: AppCompatActivity,
-  val promotedStoryList: List<PromotedStoryViewModel>
+  val promotedStoryList: List<PromotedStoryViewModel>,
+  private val recommendedActivityList: RecommendedActivityList
 ) : HomeItemViewModel() {
   private val routeToRecentlyPlayedListener = activity as RouteToRecentlyPlayedListener
 
-  // TODO(#2297): Update this span count and move to values/integers.xml once behavior is clarified
-  private val promotedStoriesTabletSpanCount: Int =
-    if (Resources.getSystem().configuration.orientation == Configuration.ORIENTATION_PORTRAIT) 2
-    else 3
-
   /** Returns the padding placed at the end of the promoted stories list based on the number of promoted stories. */
   val endPadding =
     if (promotedStoryList.size > 1)
       activity.resources.getDimensionPixelSize(R.dimen.home_padding_end)
     else activity.resources.getDimensionPixelSize(R.dimen.home_padding_start)
 
   /** Determines and returns the visibility for the "View All" button. */
+  fun getHeader(): String {
+    with(recommendedActivityList.recommendedStoryList) {
+      return when {
+        suggestedStoryList.isNotEmpty() -> {
+          if (recentlyPlayedStoryList.isNotEmpty() || olderPlayedStoryList.isNotEmpty()) {
+            activity.getString(R.string.stories_for_you)
+          } else
+            activity.getString(R.string.recommended_stories)
+        }
+        recentlyPlayedStoryList.isNotEmpty() -> {
+          activity.getString(R.string.recently_played_stories)
+        }
+        else -> {
+          activity.getString(R.string.last_played_stories)
+        }
+      }
+    }
+  }
+
+  /**
+   * Determines and returns the visibility for the "View All" button.
+   */
   fun getButtonVisibility(): Int {
```

Comment:
I suggest renaming this to be a bit clearer: ``getViewAllButtonVisibility`` per the KDoc.

Reply:
Done.